llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -860,11 +860,13 @@ RISCVTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
     break;
   }
   // TODO: add more intrinsic
-  case Intrinsic::experimental_stepvector: {
-    unsigned Cost = 1; // vid
-    auto LT = getTypeLegalizationCost(RetTy);
-    return Cost + (LT.first - 1);
-  }
+  case Intrinsic::experimental_stepvector:
+    if (ST->hasVInstructions()) {
+      auto LT = getTypeLegalizationCost(RetTy);
```

Comment:
Do we ever reach here without `+v` and need to return the scalarized cost?

Reply:
No, not typically. I just recalled the case in #83294

Reply:
If that's the case then should we still return `1 + (getTypeLegalizationCost(RetTy).First - 1)` for the scalar case so it's the same as before?

Reply:
Sure!